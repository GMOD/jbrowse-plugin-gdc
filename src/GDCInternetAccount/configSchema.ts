import { ConfigurationSchema } from '@jbrowse/core/configuration'
import { Instance } from 'mobx-state-tree'
import PluginManager from '@jbrowse/core/PluginManager'
import { createInternetAccountConfig } from '@jbrowse/core/pluggableElementTypes/models'

function GDCConfigFactory(pluginManager: PluginManager) {
  return ConfigurationSchema(
    'GDCInternetAccount',
    {
      validDomains: {
        description:
          'array of valid domains the url can contain to use this account. Empty = all domains',
        type: 'stringArray',
        defaultValue: [],
      },
      authHeader: {
        description: 'custom auth header for authorization',
        type: 'string',
        defaultValue: 'Authorization',
      },
      customEndpoint: {
        description: 'custom endpoint for the external token resource',
        type: 'string',
        defaultValue: '',
      },
    },
    {
      baseConfiguration: createInternetAccountConfig(pluginManager),
      explicitlyTyped: true,
    },
  )
}

export type GDCInternetAccountConfigModel = ReturnType<typeof GDCConfigFactory>
export type OAuthInternetAccountConfig = Instance<GDCInternetAccountConfigModel>
export default GDCConfigFactory
