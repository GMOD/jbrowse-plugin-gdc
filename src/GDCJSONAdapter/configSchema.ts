import { ConfigurationSchema } from '@jbrowse/core/configuration'
import { types } from 'mobx-state-tree'

export default ConfigurationSchema(
  'GDCJSONAdapter',
  {
    featureType: {
      type: 'stringEnum',
      model: types.enumeration('Feature Type', ['mutation', 'gene']),
      defaultValue: 'mutation',
      description: 'The type of the track to add',
    },
    data: {
      type: 'string',
      defaultValue: '',
      description: 'JSON string of the GDC data from the file'
    }
  },
  { explicitlyTyped: true },
)
