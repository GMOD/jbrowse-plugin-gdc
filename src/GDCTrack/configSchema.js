import { ConfigurationSchema } from '@gmod/jbrowse-core/configuration'

export default pluginManager => {
    const { BaseTrackConfig: LinearGenomeTrackConfig }  = pluginManager.getPlugin('LinearGenomeViewPlugin').exports
    return ConfigurationSchema(
    'GDCTrack',
    {
      adapter: pluginManager.pluggableConfigSchemaType('adapter'),
      renderer: pluginManager.pluggableConfigSchemaType('renderer'),
    },
    { baseConfiguration: LinearGenomeTrackConfig, explicitlyTyped: true },
  )
}
