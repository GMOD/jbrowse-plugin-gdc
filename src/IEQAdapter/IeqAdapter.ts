import {
  BaseFeatureDataAdapter,
  BaseOptions,
} from '@jbrowse/core/data_adapters/BaseAdapter'
import { FileLocation, Region } from '@jbrowse/core/util/types'
import { openLocation } from '@jbrowse/core/util/io'
import { ObservableCreate } from '@jbrowse/core/util/rxjs'
import IeqFeature from './IeqFeature'
import { Feature } from '@jbrowse/core/util/simpleFeature'
import { readConfObject } from '@jbrowse/core/configuration'
import { AnyConfigurationModel } from '@jbrowse/core/configuration/configurationSchema'
import PluginManager from '@jbrowse/core/PluginManager'
import { getSubAdapterType } from '@jbrowse/core/data_adapters/dataAdapterCache'

/**
 * Isoform Expression Quantification Adapter
 */
export default class IeqAdapter extends BaseFeatureDataAdapter {
  public static capabilities = ['getFeatures', 'getRefNames']

  private setupP?: Promise<Feature[]>

  private async readIeq() {
    const ieqLocation = readConfObject(
      this.config,
      'ieqLocation',
    ) as FileLocation
    const fileContents = await openLocation(
      ieqLocation,
      this.pluginManager,
    ).readFile('utf8')

    const lines = fileContents.split('\n')
    const rows: string[] = []
    let columns: string[] = []
    lines.forEach(line => {
      if (line) {
        if (columns.length === 0) {
          columns = line.split('\t')
        } else {
          rows.push(line)
        }
      }
    })

    return {
      lines: rows,
      columns,
    }
  }

  private parseCoords(property: string) {
    const splitProperty = property.split(':')
    return {
      chromosome: splitProperty[1],
      start: splitProperty[2].split('-')[0],
      end: splitProperty[2].split('-')[1],
      strand: splitProperty[3] === '+' ? 1 : 0,
    }
  }

  private parseLine(line: string, columns: string[]) {
    let iso: any = {}
    line.split('\t').forEach((property: string, i: number) => {
      if (property) {
        if (columns[i] === 'isoform_coords') {
          const parsedProperties = this.parseCoords(property)
          iso = {
            ...iso,
            ...parsedProperties,
          }
        } else {
          iso[columns[i].toLowerCase()] = property
        }
      }
    })
    return iso
  }

  private async getLines() {
    const { columns, lines } = await this.readIeq()

    return lines.map((line, index) => {
      return new IeqFeature({
        iso: this.parseLine(line, columns),
        id: `${this.id}-ieq-${index}`,
      })
    })
  }

  private async setup() {
    if (!this.setupP) {
      this.setupP = this.getLines()
    }
    return this.setupP
  }

  public async getRefNames(_: BaseOptions = {}) {
    return [
      'chr1',
      'chr2',
      'chr3',
      'chr4',
      'chr5',
      'chr6',
      'chr7',
      'chr8',
      'chr9',
      'chr10',
      'chr11',
      'chr12',
      'chr13',
      'chr14',
      'chr15',
      'chr16',
      'chr17',
      'chr18',
      'chr19',
      'chr20',
      'chr21',
      'chr22',
      'chrX',
      'chrY',
    ]
  }

  public getFeatures(region: Region, opts: BaseOptions = {}) {
    return ObservableCreate<Feature>(async observer => {
      const feats = await this.setup()
      feats.forEach(f => {
        if (
          f.get('refName') === region.refName &&
          f.get('end') > region.start &&
          f.get('start') < region.end
        ) {
          observer.next(f)
        }
      })
      observer.complete()
    }, opts.signal)
  }

  public freeResources(): void {}
}
