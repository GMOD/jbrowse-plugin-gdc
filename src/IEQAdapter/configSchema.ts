import { ConfigurationSchema } from '@jbrowse/core/configuration'

export default ConfigurationSchema(
  'IeqAdapter',
  {
    ieqLocation: {
      type: 'fileLocation',
      defaultValue: { uri: '/path/to/myfile.tsv', locationType: 'UriLocation' },
    },
  },
  { explicitlyTyped: true },
)
