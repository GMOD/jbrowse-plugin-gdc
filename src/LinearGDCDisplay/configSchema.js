import { ConfigurationSchema } from '@jbrowse/core/configuration'

export default pluginManager => {
  const { baseLinearDisplayConfigSchema } = pluginManager.getPlugin(
    'LinearGenomeViewPlugin',
  ).exports
  return ConfigurationSchema(
    'LinearGDCDisplay',
    { renderer: pluginManager.pluggableConfigSchemaType('renderer') },
    { baseConfiguration: baseLinearDisplayConfigSchema, explicitlyTyped: true },
  )
}
