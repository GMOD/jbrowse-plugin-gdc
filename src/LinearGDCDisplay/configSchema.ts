import PluginManager from '@jbrowse/core/PluginManager'
import { ConfigurationSchema } from '@jbrowse/core/configuration'
import LinearGenomeViewPlugin from '@jbrowse/plugin-linear-genome-view'

export default (pluginManager: PluginManager) => {
  const { baseLinearDisplayConfigSchema } = (
    pluginManager.getPlugin('LinearGenomeViewPlugin') as LinearGenomeViewPlugin
  ).exports
  return ConfigurationSchema(
    'LinearGDCDisplay',
    { renderer: pluginManager.pluggableConfigSchemaType('renderer') },
    { baseConfiguration: baseLinearDisplayConfigSchema, explicitlyTyped: true },
  )
}
