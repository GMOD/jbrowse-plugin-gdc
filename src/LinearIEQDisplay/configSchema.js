import { ConfigurationSchema } from '@jbrowse/core/configuration'

export default pluginManager => {
  const { baseLinearDisplayConfigSchema } = pluginManager.getPlugin(
    'LinearGenomeViewPlugin',
  ).exports
  return ConfigurationSchema(
    'LinearIEQDisplay',
    { renderer: pluginManager.pluggableConfigSchemaType('renderer') },
    { baseConfiguration: baseLinearDisplayConfigSchema, explicitlyTyped: true },
  )
}
