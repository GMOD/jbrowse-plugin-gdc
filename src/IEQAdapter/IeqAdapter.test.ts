import { toArray } from 'rxjs/operators'
import IeqAdapter from './IeqAdapter'
import configSchema from './configSchema'
import fetchMock from 'jest-fetch-mock'

fetchMock.enableMocks()

test('adapter can fetch features from ieq_test_data.txt', async () => {
  const adapter = new IeqAdapter(
    configSchema.create({
      ieqLocation: {
        localPath: require.resolve('./test_data/ieq_test_data.txt'),
      },
    }),
  )

  const features = adapter.getFeatures({
    assemblyName: 'volvox',
    refName: 'chr9',
    start: 94175962,
    end: 94175981,
  })

  const names = await adapter.getRefNames()
  expect(names).toMatchSnapshot()

  const featuresArray = await features.pipe(toArray()).toPromise()
  const featuresJsonArray = featuresArray.map(f => f.toJSON())
  expect(featuresJsonArray).toMatchSnapshot()
})
