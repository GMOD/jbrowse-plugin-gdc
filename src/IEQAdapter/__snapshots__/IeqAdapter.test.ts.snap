// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adapter can fetch features from ieq_test_data.txt 1`] = `
Array [
  "chr1",
  "chr2",
  "chr3",
  "chr4",
  "chr5",
  "chr6",
  "chr7",
  "chr8",
  "chr9",
  "chr10",
  "chr11",
  "chr12",
  "chr13",
  "chr14",
  "chr15",
  "chr16",
  "chr17",
  "chr18",
  "chr19",
  "chr20",
  "chr21",
  "chr22",
  "chrX",
  "chrY",
]
`;

exports[`adapter can fetch features from ieq_test_data.txt 2`] = `
Array [
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175977,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 1 reads",
    "read_count": "1",
    "reads_per_million_mirna_mapped": "0.748138",
    "refName": "chr9",
    "start": 94175960,
    "strand": 1,
    "uniqueId": "test-ieq-0",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175981,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 1 reads",
    "read_count": "1",
    "reads_per_million_mirna_mapped": "0.748138",
    "refName": "chr9",
    "start": 94175960,
    "strand": 1,
    "uniqueId": "test-ieq-1",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175982,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 6 reads",
    "read_count": "6",
    "reads_per_million_mirna_mapped": "4.488827",
    "refName": "chr9",
    "start": 94175960,
    "strand": 1,
    "uniqueId": "test-ieq-2",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175983,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 12 reads",
    "read_count": "12",
    "reads_per_million_mirna_mapped": "8.977655",
    "refName": "chr9",
    "start": 94175960,
    "strand": 1,
    "uniqueId": "test-ieq-3",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175984,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 2 reads",
    "read_count": "2",
    "reads_per_million_mirna_mapped": "1.496276",
    "refName": "chr9",
    "start": 94175960,
    "strand": 1,
    "uniqueId": "test-ieq-4",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175980,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 66 reads",
    "read_count": "66",
    "reads_per_million_mirna_mapped": "49.377100",
    "refName": "chr9",
    "start": 94175961,
    "strand": 1,
    "uniqueId": "test-ieq-5",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175981,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 1372 reads",
    "read_count": "1372",
    "reads_per_million_mirna_mapped": "1026.445178",
    "refName": "chr9",
    "start": 94175961,
    "strand": 1,
    "uniqueId": "test-ieq-6",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175982,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 5285 reads",
    "read_count": "5285",
    "reads_per_million_mirna_mapped": "3953.908721",
    "refName": "chr9",
    "start": 94175961,
    "strand": 1,
    "uniqueId": "test-ieq-7",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175983,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 17828 reads",
    "read_count": "17828",
    "reads_per_million_mirna_mapped": "13337.802210",
    "refName": "chr9",
    "start": 94175961,
    "strand": 1,
    "uniqueId": "test-ieq-8",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175984,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 880 reads",
    "read_count": "880",
    "reads_per_million_mirna_mapped": "658.361339",
    "refName": "chr9",
    "start": 94175961,
    "strand": 1,
    "uniqueId": "test-ieq-9",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175985,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 80 reads",
    "read_count": "80",
    "reads_per_million_mirna_mapped": "59.851031",
    "refName": "chr9",
    "start": 94175961,
    "strand": 1,
    "uniqueId": "test-ieq-10",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175991,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 1 reads",
    "read_count": "1",
    "reads_per_million_mirna_mapped": "0.748138",
    "refName": "chr9",
    "start": 94175961,
    "strand": 1,
    "uniqueId": "test-ieq-11",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175980,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 3 reads",
    "read_count": "3",
    "reads_per_million_mirna_mapped": "2.244414",
    "refName": "chr9",
    "start": 94175962,
    "strand": 1,
    "uniqueId": "test-ieq-12",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175981,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 33 reads",
    "read_count": "33",
    "reads_per_million_mirna_mapped": "24.688550",
    "refName": "chr9",
    "start": 94175962,
    "strand": 1,
    "uniqueId": "test-ieq-13",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175982,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 110 reads",
    "read_count": "110",
    "reads_per_million_mirna_mapped": "82.295167",
    "refName": "chr9",
    "start": 94175962,
    "strand": 1,
    "uniqueId": "test-ieq-14",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175983,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 474 reads",
    "read_count": "474",
    "reads_per_million_mirna_mapped": "354.617357",
    "refName": "chr9",
    "start": 94175962,
    "strand": 1,
    "uniqueId": "test-ieq-15",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175984,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 25 reads",
    "read_count": "25",
    "reads_per_million_mirna_mapped": "18.703447",
    "refName": "chr9",
    "start": 94175962,
    "strand": 1,
    "uniqueId": "test-ieq-16",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175985,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 3 reads",
    "read_count": "3",
    "reads_per_million_mirna_mapped": "2.244414",
    "refName": "chr9",
    "start": 94175962,
    "strand": 1,
    "uniqueId": "test-ieq-17",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175981,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 2 reads",
    "read_count": "2",
    "reads_per_million_mirna_mapped": "1.496276",
    "refName": "chr9",
    "start": 94175963,
    "strand": 1,
    "uniqueId": "test-ieq-18",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175982,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 8 reads",
    "read_count": "8",
    "reads_per_million_mirna_mapped": "5.985103",
    "refName": "chr9",
    "start": 94175963,
    "strand": 1,
    "uniqueId": "test-ieq-19",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175983,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 38 reads",
    "read_count": "38",
    "reads_per_million_mirna_mapped": "28.429240",
    "refName": "chr9",
    "start": 94175963,
    "strand": 1,
    "uniqueId": "test-ieq-20",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175984,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 3 reads",
    "read_count": "3",
    "reads_per_million_mirna_mapped": "2.244414",
    "refName": "chr9",
    "start": 94175963,
    "strand": 1,
    "uniqueId": "test-ieq-21",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175985,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 1 reads",
    "read_count": "1",
    "reads_per_million_mirna_mapped": "0.748138",
    "refName": "chr9",
    "start": 94175963,
    "strand": 1,
    "uniqueId": "test-ieq-22",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175981,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 1 reads",
    "read_count": "1",
    "reads_per_million_mirna_mapped": "0.748138",
    "refName": "chr9",
    "start": 94175964,
    "strand": 1,
    "uniqueId": "test-ieq-23",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175982,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 1 reads",
    "read_count": "1",
    "reads_per_million_mirna_mapped": "0.748138",
    "refName": "chr9",
    "start": 94175964,
    "strand": 1,
    "uniqueId": "test-ieq-24",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175983,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 2 reads",
    "read_count": "2",
    "reads_per_million_mirna_mapped": "1.496276",
    "refName": "chr9",
    "start": 94175964,
    "strand": 1,
    "uniqueId": "test-ieq-25",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175984,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 3 reads",
    "read_count": "3",
    "reads_per_million_mirna_mapped": "2.244414",
    "refName": "chr9",
    "start": 94175964,
    "strand": 1,
    "uniqueId": "test-ieq-26",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175983,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 2 reads",
    "read_count": "2",
    "reads_per_million_mirna_mapped": "1.496276",
    "refName": "chr9",
    "start": 94175965,
    "strand": 1,
    "uniqueId": "test-ieq-27",
  },
  Object {
    "chromosome": "chr9",
    "cross-mapped": "N",
    "end": 94175983,
    "mirna_id": "hsa-let-7a-1",
    "mirna_region": "mature,MIMAT0000062",
    "name": "hsa-let-7a-1, 1 reads",
    "read_count": "1",
    "reads_per_million_mirna_mapped": "0.748138",
    "refName": "chr9",
    "start": 94175966,
    "strand": 1,
    "uniqueId": "test-ieq-28",
  },
]
`;
